crates/scheduler/src/call/uart.rs

```diff
@@ -87,7 +87,7 @@ fn read<B: Board>(mut call: SchedulerCall<B, api::read::Sig>) {
     let result = try {
         let uart = Id::new(*uart as usize).map_err(|_| Trap)?;
         let output = memory.get_mut(*ptr, *len)?;
-        board::Uart::<B>::read(uart, output).map(|x| x as u32)
+        board::Uart::<B>::read(uart, output).map(|x| x as u32)?
```

Comment:
we can also do the trick here (and in the rest of the file)
```suggestion
        board::Uart::<B>::read(uart, output)? as u32
```